app/clients/salesforce/salesforce_engagement.py

```diff
@@ -104,8 +104,8 @@ def update(
 
         # Existing Engagement, update the stage name
         if engagement:
-            result = session.Opportunity.update(
-                engagement.get("Id"),
+            result = session.Opportunity.update( # type: ignore
```

Comment:
if we are ignoring this - why do we need to convert to str? I think we need to test this on staging to see saleforce doesnt fail with these inputs.

Reply:
There's a couple of type checking issues on the `session.[SalesforceObject].[CRUD_method]` calls. The `Object` resolves to one of: `SFBulkType`, `SFBulk2Type`, or `SFType` under the hood. 

```
app/clients/salesforce/salesforce_engagement.py:107: error: "SFBulkType" not callable  [operator]
app/clients/salesforce/salesforce_engagement.py:107: error: "SFBulk2Type" not callable  [operator]
```
[Full list of errors](https://github.com/cds-snc/notification-api/actions/runs/9097826965/job/25006575947)

To get mypy to not complain on these calls we would have to [create our own SFObject](https://simple-salesforce.readthedocs.io/en/latest/user_guide/additional_features.html) and call the desired `crud` method on that. It seemed like too much of a refactor to attempt here.